source/slang/slang-ir-simplify-cfg.cpp

```diff
@@ -490,10 +490,17 @@ static bool trySimplifyIfElse(IRBuilder& builder, IRIfElse* ifElseInst)
     bool isFalseBranchTrivial = false;
     if (isTrivialIfElse(ifElseInst, isTrueBranchTrivial, isFalseBranchTrivial))
     {
-        // If both branches of `if-else` are trivial jumps into after block,
+        // If either branch of `if-else` is a trivial jump into after block,
         // we can get rid of the entire conditional branch and replace it
         // with a jump into the after block.
-        if (auto termInst = as<IRUnconditionalBranch>(ifElseInst->getTrueBlock()->getTerminator()))
+        IRUnconditionalBranch* termInst;
+        if (termInst = as<IRUnconditionalBranch>(ifElseInst->getTrueBlock()->getTerminator()))
```

Comment:
style: fold the initialization to the declaration site of termInst, and just do `if (termInst & termInst->getTargetBlock() != ifElseInst->getAfterBlock())`